### DISCUSSION TITLE
Navigation - Dynamic Subdomain (multi-tenant) for 1 single navigation structure

### DISCUSSION THREAD

**Discussion by weburnit** [area17/twill#1225] [General]
## Summary
Currently, it looks like we have to declare statically twill-navigation.php
If we have thousands of tenants, the current approach is not compatible. 

## Describe the solution you'd like
We might need to support one configuration for navigation only.

**Comment by ifox** [comment]
Hi @weburnit, you can actually define the navigation dynamically, in a Laravel service provider, so you could register the same config for all your tenants.

**Comment by weburnit** [comment]
```php
//config/twill-navigation.php

<?php

$configs = [
    'paul' => [ // tenant-key on sub domain: admin.paul.mydomain.com
        'campaigns' => [
            'title'  => 'Campaigns',
            'module' => true,
        ],
    ]
];

return $configs;
```

I don't know how to make $config really work dynamically for your idea!
Can you elaborate more?

**Comment by weburnit** [comment]
> Hi @weburnit, you can actually define the navigation dynamically, in a Laravel service provider, so you could register the same config for all your tenants.

```php

    'support_subdomain_admin_routing' => true,
    'admin_app_subdomain'             => 'admin',

```
```php
//previous navigation
return [
        'campaigns' => [
            'title'  => 'Campaigns',
            'module' => true,
        ],
];
```
I did enable subdomain routing for multi-tenant! previous Navigation configuration causes an error!

**Comment by ifox** [comment]
Once you enable subdomain routing, the configuration is expecting to be keyed by the subdomains. If you do not want to do this manually from the config file, you can do it dynamically by setting the config using `config(['key' => 'value'])` or `Config::set(...)`, for example.

**Comment by weburnit** [comment]
If we have a million tenants, does this solution have any impacts regarding config.cache?

I think the config should be consistent and isolated by the user's context!
That would be great to open navigation by internal design(interface) rather than config()
